### PR TITLE
Upgrade python to Python 3.8

### PR DESCRIPTION
**Pull request by jon-betts**
For: https://github.com/hypothesis/lms/issues/2590

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -1,66 +1,108 @@
...
+      python_headline_version: ${{ steps.list-python-versions.outputs.python_headline_version }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
```
Comment (LMS007) [inline .github/workflows/continuous-integration.yml:17]: I take it 3.8.9 was not an option in github actions?

**Review by LMS007 — COMMENTED** [review]
I did run `brew upgrade`  and I installed python 3.8.9 in pyenv on my mac, but I was not able to start the app with 3.8.9. I'm not really sure what went wrong because the errors are numerous.  I have saved dev-2.log locally in case its helpful to look at? but I can't make much sense of it.

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (jon-betts) [reply .github/workflows/continuous-integration.yml:17]: It's an option, but we don't need to run on an older version of Python for this stage. So it's one less thing to try and fix up.

**Comment by jon-betts** [comment]
Hi @LMS007, yeah it might be a good idea to wait until we've worked through this before diving in, but we do need to get this working for you. I'm going to be honest, I'm not really sure what `brew` is...

This is currently failing on some coverage by the looks of it, but this is very much enough for today.

**Comment by LMS007** [comment]
> Hi @LMS007, yeah it might be a good idea to wait until we've worked through this before diving in, but we do need to get this working for you. I'm going to be honest, I'm not really sure what `brew` is...
> 
> This is currently failing on some coverage by the looks of it, but this is very much enough for today.

Fair enough, brew is more or less a a way to install these packages on mac and make them somewhat consistent for everyone. I think of it a a wrapper around chaos, but it does mean that things do work a bit differently on mac. it may just mean I need upgrade or change some installed packages as we hit errors.

**Review by marcospri — APPROVED** [review]

**Pull request merged**